ExperimentTwitter/User+Twitter.m

```diff
@@ -0,0 +1,83 @@
+//
+//  User+Twitter.m
+//  ExperimentTwitter
+//
+//  Created by ayur.j on 11/09/17.
+//  Copyright © 2017 ayur.j. All rights reserved.
+//
+
+#import "User+Twitter.h"
+#import "TwitterClient.h"
+#import "CoreDataHelper.h"
+
+NSString * const UserDidLogInNotification = @"UserDidLoginNotification";
+NSString * const UserDidLogOutNotification = @"UserDidLogoutNotification";
+
+@interface User()
+@end
+
+
+@implementation User (Twitter)
+
++(User*) userWithTwitterInfo:(NSDictionary *)dictionary inManagedObjectContext:(NSManagedObjectContext *)context
+{
+    User *user = nil;
+    NSString* userId = dictionary[@"id_str"];
+    NSFetchRequest *request = [NSFetchRequest fetchRequestWithEntityName:@"User"];
+    request.predicate = [NSPredicate predicateWithFormat:@"userId = %@", userId];
+    NSError *error;
+    NSArray *matches = [context executeFetchRequest:request error:&error];
+    if(!matches || error || [matches count] > 1) {
+       //
+    } else if([matches count]) {
+        user = [matches firstObject];
+    } else {
+        user = [NSEntityDescription insertNewObjectForEntityForName:@"User" inManagedObjectContext:context];
+        user.name = dictionary[@"name"];
+        user.screenName = dictionary[@"screen_name"];
+        user.profileImageUrl = dictionary[@"profile_image_url_https"];
+        user.userId = dictionary[@"id_str"];
+        [CoreDataHelper saveManagedObjectContext];
+    }
+    return user;
+}
+
+static User *_currentUser = nil;
```

Comment:
`@synthesize` instead

Reply:
Compiler says @synthesize is not allowed in category

Reply:
Oh right. Didn't realize it's a category.